code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

```diff
@@ -104,70 +105,105 @@ Difficulty: Medium
 /mob/living/simple_animal/hostile/megafauna/dragon/OpenFire()
 	if(swooping)
 		return
+	if(swoop_cooldown >= world.time)
+		return
 	anger_modifier = CLAMP(((maxHealth - health)/50),0,20)
 	ranged_cooldown = world.time + ranged_cooldown_time
-
+	
 	if(prob(15 + anger_modifier) && !client)
 		if(health < maxHealth/2)
-			INVOKE_ASYNC(src, .proc/swoop_attack, TRUE, null, 50)
+			INVOKE_ASYNC(src, .proc/swoop_attack, FALSE)
 		else
-			fire_rain()
+			lava_pools()
 
 	else if(prob(10+anger_modifier) && !client)
 		if(health > maxHealth/2)
 			INVOKE_ASYNC(src, .proc/swoop_attack)
 		else
 			INVOKE_ASYNC(src, .proc/triple_swoop)
 	else
-		fire_walls()
+		fire_cone()
 
-/mob/living/simple_animal/hostile/megafauna/dragon/proc/fire_rain()
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/lava_pools()
 	if(!target)
 		return
-	target.visible_message("<span class='boldwarning'>Fire rains from the sky!</span>")
-	for(var/turf/turf in range(9,get_turf(target)))
-		if(prob(11))
+	target.visible_message("<span class='boldwarning'>Lava starts to pool up beneath you!</span>")
+	for(var/turf/turf in range(10,get_turf(target)))
+		if(prob(35))
 			new /obj/effect/temp_visual/target(turf)
```

Comment:
This is 441 checks, resulting in ~130 turf change operations on average, each of which has an icon change and tries to ignite every item on top, making them process the fire damage. This is expensive on its own, and worse if it's spammable by the drake.

Reply:
I might rework this attack altogether because any less of a range and the player can just completely run away from the attack through the turning lava.

Reply:
Does this new attack solve the problem?